binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/test/java/org/springframework/cloud/stream/binder/reactorkafka/ReactorKafkaBinderIntegrationTests.java

```diff
@@ -107,34 +109,30 @@ void endToEndReactorKafkaBinder(boolean excludeKafkaAutoConfig) {
 				"--spring.cloud.stream.bindings.lowercase-out-0.destination=lowercased-words",
 				"--spring.cloud.stream.bindings.patternConsumer-in-0.group=grp3",
 				"--spring.cloud.stream.bindings.patternConsumer-in-0.destination=.*-words",
+				"--spring.cloud.stream.kafka.bindings.words1.producer.record-metadata-channel=sendResults",
+				"--spring.cloud.stream.kafka.bindings.words2.producer.record-metadata-channel=sendResults",
 				"--spring.cloud.stream.kafka.bindings.patternConsumer-in-0.consumer.destination-is-pattern=true",
 				"--spring.cloud.stream.kafka.bindings.lowercase-in-0.consumer.converterBeanName=fullRR",
 				"--spring.cloud.stream.kafka.binder.brokers=" + embeddedKafka.getBrokersAsString(),
 				excludeKafkaAutoConfigParam(excludeKafkaAutoConfig))) {
 
-			Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
-			DefaultKafkaProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
-			try {
-				KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf, true);
-				template.send("words1", "foobar");
-				template.send("words2", "BAZQUX");
-
-				assertThat(KafkaTestUtils.getSingleRecord(consumer1, "uppercased-words"))
-					.isNotNull()
-					.extracting(ConsumerRecord::value)
-					.isEqualTo("FOOBAR");
-
-				assertThat(KafkaTestUtils.getSingleRecord(consumer2, "lowercased-words"))
-					.isNotNull()
-					.extracting(ConsumerRecord::value)
-					.isEqualTo("bazqux");
-
-				assertThat(recOptsCustOrder).containsExactly("two", "one", "two", "one", "two", "one");
-				await().untilAsserted(() -> assertThat(patternedDeliveries).contains("bazqux", "FOOBAR"));
-			}
-			finally {
-				pf.destroy();
-			}
+			StreamBridge streamBridge = context.getBean(StreamBridge.class);
+			streamBridge.send("words1", MessageBuilder.withPayload("foobar").setCorrelationId(42).build());
+			streamBridge.send("words2", MessageBuilder.withPayload("BAZQUX").setCorrelationId(43).build());
+
+			assertThat(KafkaTestUtils.getSingleRecord(consumer1, "uppercased-words"))
+				.isNotNull()
```

Comment:
**[INFO-SHARING]** The `isNotNull()` in many cases is redundant as almost all of the other AssertJ assertions all do a `isNotNull` before anything else. You may already know this and prefer the `isNotNull` here to be explicit. Just sharing in case you did not know about this.

Reply:
Thanks for the tip, I was not aware, but I do think I prefer the explicit method call.

Reply:
The other interesting AssertJ redundant check (that is more valuable and does not provide much value being there explicitly) is the context assert when used in say `ApplicationContextRunner`. 
```java
assertThat(context)
  .hasNotFailed()
  .hasSingleBean("foo").hasBlah()....
```
It is somewhat expected that the context has not failed and when you see it 100 times in 100 tests in a single file it is a bit redundant. 

Anyways, in case you run into that one.